src/utils/victim-identification/victim.identifier.ts

```diff
@@ -0,0 +1,361 @@
+          // Determine whether 'from' and 'to' are contracts
+          let isFromContract = this.isContractCache.get(from);
+          if (isFromContract === undefined) {
+            isFromContract = (await this.provider.getCode(from)) !== "0x";
+            if (isFromContract) {
+              this.isContractCache.set(from, true);
+            }
+          }
+          let isToContract = this.isContractCache.get(to);
+          if (isToContract === undefined) {
+            isToContract = (await this.provider.getCode(to)) !== "0x";
+            if (isToContract) {
+              this.isContractCache.set(to, true);
+            }
+          }
+          // Update the native token balance for the from address
+          if (balanceChangesMap.has(from)) {
+            let currentEntry = balanceChangesMap.get(from);
+            currentEntry!["native"] = (currentEntry!["native"] || ZERO).sub(bnValue);
+            balanceChangesMap.set(from, currentEntry!);
+          } else if (isFromContract) {
+            balanceChangesMap.set(from, { ["native"]: bnValue.mul(-1) });
+          }
+          // Update the native token balance for the to address
+          if (balanceChangesMap.has(to)) {
+            let currentEntry = balanceChangesMap.get(to);
+            currentEntry!["native"] = (currentEntry!["native"] || ZERO).add(bnValue);
+            balanceChangesMap.set(to, currentEntry!);
+          } else if (isToContract) {
+            balanceChangesMap.set(to, { ["native"]: bnValue });
+          }
+        }
+      })
+    );
+
+    // Remove empty records
+    balanceChangesMap.forEach((record: Record<string, ethers.BigNumber>, key: string) => {
+      Object.keys(record).forEach((token) => {
+        if (record[token].eq(ZERO)) {
+          delete record[token];
+        }
+      });
+      if (Object.keys(record).length === 0) {
+        balanceChangesMap.delete(key);
+      }
+    });
+    const balanceChangesMapUsd: Map<string, Record<string, number>> = new Map();
+    // Get the USD value of the balance changes
+    await Promise.all(
+      Array.from(balanceChangesMap.entries()).map(async ([key, record]) => {
+        const usdRecord: Record<string, number> = {};
+        await Promise.all(
+          Object.keys(record).map(async (token) => {
+            const UsdValue = await this.getValueInUsd(
+              txEvent.blockNumber,
+              txEvent.network,
+              record[token].toString(),
+              token
+            );
+            usdRecord[token] = UsdValue;
+          })
+        );
+        balanceChangesMapUsd.set(key, usdRecord);
+      })
+    );
+
+    const victims: string[] = [];
+    balanceChangesMapUsd.forEach((record: Record<string, number>, key: string) => {
+      const sum = Object.values(record).reduce((acc, value) => {
+        return acc + value;
+      }, 0);
+      // If the sum of the values is less than -100 USD, add the address to the victims list
+      if (sum < -100) {
```

Comment:
This is up for discussion. The 3 choices that I've considered and tested were -1, -10, -100 but the first two are too small, imo, for an address to be considered a victim.

Not sure if you think it can be bigger or smaller though.

Reply:
I think it can stay at `-100`. 👍 

Mostly wanted to know how this specific figure was arrived at